src/main/java/com/marklogic/appdeployer/command/groups/DeployGroupsCommand.java

```diff
@@ -35,11 +40,21 @@ protected ResourceManager getResourceManager(CommandContext context) {
     protected void afterResourceSaved(ResourceManager mgr, CommandContext context, File resourceFile,
             SaveReceipt receipt) {
         String payload = receipt.getPayload();
-        if (payload != null && payload.contains("cache-size") && context.getAdminManager() != null) {
-            if (logger.isDebugEnabled()) {
-                logger.info("Group payload contains cache-size parameter, so waiting for ML to restart");
-            }
-            context.getAdminManager().waitForRestart();
+        if (payload != null) {
+        	if (payload.contains("cache-size") && context.getAdminManager() != null) {
+                if (logger.isDebugEnabled()) {
+                    logger.info("Group payload contains cache-size parameter, so waiting for ML to restart");
+                }
+                context.getAdminManager().waitForRestart();
+        	}
+        	Fragment xml = new Fragment(payload);
+        	String groupName = xml.getElementValue("/m:group-properties/m:group-name");
+
+			// When new groups are created, an Admin server is automatically created in that group.
+			// However, the Admin server's rewrite property is empty - causing problems with reading the timestamp
+            String adminServerPayload = format(ADMIN_SERVER_UPDATE_JSON_TEMPLATE, ADMIN_SERVER_NAME, groupName);
```

Comment:
I forgot to mention this before - there's a class for each resource, and those classes have a getJson method. So you can do:

    Server s = new Server(null, "Admin"); // the API param isn't needed for this
    s.setUrlRewriter("whatever");
    s.getJson();

I always like for commands to have setters for things that could conceivably change. So I'm thinking this - declare a private variable of "private Server adminServerTemplate", which the constructor for this command initializes with a server-name of "Admin" and the appropriate url-rewriter value. Then add a setter for it so that a client can always modify how the Admin object is created (e.g. if the user wants to see a different authentication value on it).